eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```
@@ -1380,60 +1406,65 @@ public HealthCheckHandler getHealthCheckHandler() {
+                lastSuccessfulRegistryFetchTimestamp = System.currentTimeMillis();
+            }
 
-                if (logger.isDebugEnabled()) {
-                    StringBuilder allAppsHashCodes = new StringBuilder();
-                    allAppsHashCodes.append("Local region apps hashcode: ");
-                    allAppsHashCodes.append(localRegionApps.get().getAppsHashCode());
-                    allAppsHashCodes.append(", is fetching remote regions? ");
-                    allAppsHashCodes.append(isFetchingRemoteRegionRegistries);
-                    for (Map.Entry<String, Applications> entry : remoteRegionVsApps.entrySet()) {
-                        allAppsHashCodes.append(", Remote region: ");
-                        allAppsHashCodes.append(entry.getKey());
-                        allAppsHashCodes.append(" , apps hashcode: ");
-                        allAppsHashCodes.append(entry.getValue().getAppsHashCode());
-                    }
-                    logger.debug("Completed cache refresh task for discovery. All Apps hash code is {} ",
-                            allAppsHashCodes.toString());
+            if (logger.isDebugEnabled()) {
+                StringBuilder allAppsHashCodes = new StringBuilder();
+                allAppsHashCodes.append("Local region apps hashcode: ");
+                allAppsHashCodes.append(localRegionApps.get().getAppsHashCode());
+                allAppsHashCodes.append(", is fetching remote regions? ");
+                allAppsHashCodes.append(isFetchingRemoteRegionRegistries);
+                for (Map.Entry<String, Applications> entry : remoteRegionVsApps.entrySet()) {
+                    allAppsHashCodes.append(", Remote region: ");
+                    allAppsHashCodes.append(entry.getKey());
+                    allAppsHashCodes.append(" , apps hashcode: ");
+                    allAppsHashCodes.append(entry.getValue().getAppsHashCode());
                 }
-            } catch (Throwable th) {
```

As this is thread outermost loop try/catch we should catch Throwable exceptions. Otherwise the registry refresher may silently fail without us knowing about it.